src/basic/pidref.c

```diff
@@ -46,6 +46,10 @@ bool pidref_equal(PidRef *a, PidRef *b) {
         if (!pidref_is_set(b))
                 return false;
 
+        /* If this is the very same structure, it definitely refers to the same process */
+        if (a == b)
```

Comment:
Why not check this at the very beginning (but probably with `_unlikely_`)

Reply:
i can move this up. but not convinced of the `_unlikely_` thing. You never know how this ends up being called. This is generic code and it's very hard to know for sure that we won't end up calling this in a tight loop with two equal arguments.

i think `_unlikely_` is fine to use for stuff that doesn't have much external input, i.e. that implements something very specific, and you hence know how things are called. but this is really not like that. A misplaced likely/unlikely is definitely worse than none at all...

Reply:
moved it up now, but didn't to the unlikely thing for the mentioned reasons